go.mod

```diff
@@ -42,6 +43,7 @@ require (
 	github.com/CycloneDX/cyclonedx-go v0.7.0
 	github.com/DataDog/agent-payload/v5 v5.0.81
 	github.com/DataDog/appsec-internal-go v0.0.0-20230215162203-5149228be86a
+	github.com/DataDog/datadog-agent/pkg/gohai v0.0.0-20230512151819-dfebac949c96
```

Comment:
I discussed it with agent-platform because I wasn't sure how it works, gohai was added to the module list in `tasks/modules.py` so:
- a tag will be created automatically at each release
- the version in `go.mod` will be updated to the last tag every time it happens
- the agent always uses the last commit (due to `replace github.com/Datadog/datadog-agent/pkg/gohai => ./pkg/gohai` above)

The version tag in this file exists to pin a precise version if the agent itself is used as a dependency.
If all goes well we have nothing to do about it.